zerver/models.py

```diff
@@ -1059,6 +1061,39 @@ def parse_usermessage_flags(val):
         mask <<= 1
     return flags
 
+class Attachment(models.Model):
+    MAX_FILENAME_LENGTH = 100
+    file_name = models.CharField(max_length=MAX_FILENAME_LENGTH, db_index=True)
+    # path_id is a storage location agnostic representation of the path of the file.
+    # If the path of a file is http://localhost:9991/user_uploads/a/b/abc/temp_file.py
+    # then its path_id will be /a/b/abc/temp_file.py.
+    path_id = models.TextField(db_index=True)
+    owner = models.ForeignKey(UserProfile)
+    message = models.ForeignKey(Message, blank=True, null=True)
```

Comment:
I've been thinking about this more, and I think we should replace `message` with `messages` (and make it a ManyToMany field).  This lets us solve the "claiming an already claimed" message problem we made a TODO about earlier.  When a new message tries to claim a file that's already been claimed, we should just add that message to the list of messages connected to the attachment.

And later it'll make it possible for us to correctly implement garbage collection if one of the messages that reference an attachment is later garbage-collected due to a message retention policy (#106).